packages/dds/tree/docs/main/compatibility.md

```diff
@@ -147,3 +147,83 @@ Snapshot tests are effective at catching changes which inadvertently modify the
 
 Tree2's full-scale snapshot tests can be found at [experimental/dds/tree2/src/test/snapshots/summary.spec.ts](../../src/test/snapshots/summary.spec.ts),
 with smaller-scale snapshot tests (e.g. snapshot testing just the SchemaIndex format) nearby.
+
+# Implementation Specifics
+
+SharedTree's codecs are frequently composed over each other in a manner consistent with SharedTree's layering.
+This conveniently allows unit testing only portions of the persisted data.
+As a downside, it makes maintaining backwards compatibility with the format somewhat more complex, because not all codecs in this composition provide explicit versions for the portion of data they encode.
+The following diagram shows _runtime dependencies_ of the codec hierarchy for SharedTree's original persisted format.
+
+```mermaid
+flowchart TD
```

Comment:
This diagram will need an entry for DetachedFieldIndex

Reply:
There is one under `DetachedNodeToFieldCodec`. Maybe adds some credence other comment though :)